contracts/src/scripts/DeployGovernance.s.sol

```diff
@@ -0,0 +1,204 @@
+            type(Governance).creationCode,
+            abi.encode(
+                address(lqty),
+                address(boldToken),
+                stakingV1,
+                address(boldToken),
+                governanceConfiguration,
+                initialInitiatives
+            )
+        );
+        address governanceAddress = vm.computeCreate2Address(_salt, keccak256(bytecode));
+
+        return (governanceAddress, governanceConfiguration);
+    }
+
+    function deployEnvironment(IERC20 _boldToken, IERC20 _usdc) private {
+        boldToken = _boldToken;
+        usdc = _usdc;
+        lqty = new ERC20Faucet("Liquity", "LQTY", 100 ether, 1 days);
+        stakingV1 = address(new MockStakingV1(address(lqty)));
+    }
+
+    function deployUniV4Donations(IGovernance _governance) private {
+        uint160 flags = uint160(Hooks.AFTER_INITIALIZE_FLAG | Hooks.AFTER_ADD_LIQUIDITY_FLAG);
+
+        (, bytes32 salt) = HookMiner.find(
+            0x4e59b44847b379578588920cA78FbF26c0B4956C,
+            // address(this),
+            flags,
+            type(UniV4Donations).creationCode,
+            abi.encode(
+                address(_governance),
+                address(boldToken),
+                address(lqty),
+                block.timestamp,
+                EPOCH_DURATION,
+                address(poolManager),
+                address(usdc),
+                FEE,
+                MAX_TICK_SPACING
+            )
+        );
+
+        uniV4Donations = new UniV4Donations{salt: salt}(
+            address(_governance),
+            address(boldToken),
+            address(lqty),
+            block.timestamp,
+            EPOCH_DURATION,
+            address(poolManager),
+            address(usdc),
+            FEE,
+            MAX_TICK_SPACING
+        );
+
+        assert(address(governance) == address(uniV4Donations.governance()));
+        initialInitiatives.push(address(uniV4Donations));
+    }
+
+    function deployCurveV2GaugeRewards(IGovernance _governance) private {
+        address[] memory _coins = new address[](2);
+        _coins[0] = address(boldToken);
+        _coins[1] = address(usdc);
+        uint8[] memory _asset_types = new uint8[](2);
+        _asset_types[0] = 0;
+        _asset_types[1] = 0;
+        bytes4[] memory _method_ids = new bytes4[](2);
+        _method_ids[0] = 0x0;
+        _method_ids[1] = 0x0;
+        address[] memory _oracles = new address[](2);
+        _oracles[0] = address(0x0);
+        _oracles[1] = address(0x0);
+
+        curvePool = ICurveStableswapNG(
+            curveFactory.deploy_plain_pool(
+                "BOLD-USDC", "BOLDUSDC", _coins, 200, 1000000, 50000000000, 866, 0, _asset_types, _method_ids, _oracles
+            )
+        );
+
+        gauge = ILiquidityGauge(curveFactory.deploy_gauge(address(curvePool)));
+
+        curveV2GaugeRewards =
+            new CurveV2GaugeRewards(address(_governance), address(boldToken), address(lqty), address(gauge), DURATION);
```

Comment:
For `CurveV2GaugeRewards` to work, BOLD has to be added as a reward token to the gauge. In ChickenBonds, we made this work by deploying the gauge through the "Gauge Manager Proxy" (https://etherscan.io/address/0xd05Ad7fb0CDb39AaAA1407564dad0EC78d30d564#code), which allows the deployer to call `add_reward()` indirectly, through the proxy itself. I don't know if there's a Sepolia equivalent.

Reply:
This is supposed to be easier now:
https://docs.curve.fi/curve_dao/ownership-proxy/GaugeProxy/?h=gauge#permissionless-rewards

But I haven’t managed to make it work on Sepolia yet.